src/generic/GenericTypes.jl

```diff
@@ -509,6 +509,38 @@ mutable struct SparsePoly{T <: RingElement} <: AbstractAlgebra.RingElem
                                                new{T}([a], [UInt(0)], 1)
 end
 
+
+###############################################################################
+#
+#   LaurentPolyRing / LaurentPoly
+#
+###############################################################################
+
+abstract type LaurentPolynomialRing{T} <: AbstractAlgebra.LaurentPolynomialRing{T} end
+
+struct LaurentPolyWrapRing{T  <: RingElement,
+                           PR <: AbstractAlgebra.PolyRing{T}
+                          } <: LaurentPolynomialRing{T}
+   polyring::PR
+
+   function LaurentPolyWrapRing(pr::PR) where {T <: RingElement,
+                                               PR <: AbstractAlgebra.PolyRing{T}}
+      new{T, PR}(pr)
+   end
+end
+
+mutable struct LaurentPolyWrap{T  <: RingElement,
+                               PE <: AbstractAlgebra.PolyElem{T}
+                              } <: AbstractAlgebra.LaurentPolyElem{T}
+   poly::PE
+   mindeg::Int
```

Comment:
(Note that in principle there are also Laurent rings in more than one variable... but I'd not bother with that until somebody has an actual application...)

Reply:
Yes I was considering picking on implementing the multivariate version, but not before this one is merged and indeed someone suggests it would be useful (Uli said in an email that "one variable is sufficient for most purposes at the moment", which suggests a possible eventual use case for the multivariate version).

Reply:
The univariate Laurent polynomial ring is something you'd use all the time in (graded) representation theory, so it's great and very useful to have. Multivariate rings certainly have their applications, e.g. as base rings of (cyclotomic) Hecke algebras and in general in some geometric and combinatorics contexts; but I wouldn't make this a priority right now.